src/js/Car.js

```diff
@@ -0,0 +1,54 @@
+const CarStatus = Object.freeze({
+  STOP: 'STOP',
+  MOVING: 'MOVING',
+});
+
+function moveableNumber() {
+  const MOVABLE_RANGE_MIN_NUMBER = 0;
+  const MOVABLE_RANGE_MAX_NUMBER = 9;
+  return (
+    Math.random() * (MOVABLE_RANGE_MAX_NUMBER - MOVABLE_RANGE_MIN_NUMBER) +
+    MOVABLE_RANGE_MIN_NUMBER
+  );
+}
+
+export class Car {
+  #name;
+  #status;
+  #target;
+
+  constructor({ name, target }) {
+    this.#name = name;
+    this.#target = target;
+    this.#stop();
+  }
+
+  #move() {
+    this.#status = CarStatus.MOVING;
+  }
+
+  #stop() {
+    this.#status = CarStatus.STOP;
+  }
+
+  get $target() {
+    return this.#target;
+  }
+
+  isMoveStatus() {
+    return this.#status === CarStatus.MOVING;
+  }
+
+  static #isEnableMove(number) {
+    const MOVABLE_NUMBER = 4;
+    return number >= MOVABLE_NUMBER;
+  }
+
+  run() {
+    if (Car.#isEnableMove(moveableNumber())) {
+      this.#move();
+      return;
+    }
+    this.#stop();
+  }
```

Comment:
이 부분은 어느정도 정답(?)에 가까운 가이드를 드려보자면
```suggestion
  run(movingStrategy) {
    if (!(movingStrategy instanceOf MovingStrategy)) throw new Error('인자로 MovingStrategy를 받아와야 합니다.');
    return movingStrategy.isMoveable() ? this.#move() : this.#stop();
  }
```

이렇게 표현해주면 어떨까요?
전진 조건을 함수로 주입해주는거죠!
이렇게 할 경우 유연하게 사용할 수 있답니다 ㅎㅎ

이를 객체지향 관점으로 더 발전시켜보자면

class MovingStrategy {
  isMoveable() {
    throw new Error("MovingStrategy는 추상 클래스입니다. 별도의 구현이 필요합니다.");
  }
}

// application에서 사용되는 클래스
// 이걸 싱글톤으로 만들어서 사용할 수도 있겠죠?
class RandomMovingStrategy extends MovingStrategy {
  isMoveable() {
    return Math.random() * 10 > 4;
  }
}

// 나중에 테스트 할 때는 이런 클래스를 만들어서 사용해보세요
class AlwaysMovingStrategy extends MovingStrategy {
  isMoveable() {
    return true;
  }
}
class NotMovingStrategy extends MovingStrategy {
  isMoveable() {
    return false;
  }
}
class ToggleMovingStrategy extends MovingStrategy {
  toggle = false;
  isMoveable() {
    this.toggle = !this.toggle;
    return this.toggle;
  }
}
이럴 경우 자동차가 움직이는 조건 자체를 관리할 수 있게 됩니다.
뿐만아니라 전진 조건도 별도로 테스트가 가능하고, 자동차 자체에 대한 테스트도 가능해지겠죠!?

feat: 참고로 위와 같은 행위를 전략패턴 이라고 합니다.

Reply:
아하 
그러면 `Car.run`을 호출 할 때 인자로 `RandomMovingStrategy` 를 생성하여 넘겨주면 되겠네요.
테스트 할 때는 바꿔서 넘겨주면 되겠군요. 
감사합니다.